src/RegionCosts.cpp

```diff
@@ -0,0 +1,622 @@
+        }
+    }
+
+    internal_assert(region_cost.arith != unknown && region_cost.memory != unknown);
+    return region_cost;
+}
+
+Cost RegionCosts::region_cost(const map<string, Box> &regions, const set<string> &inlines){
+    Cost total_cost(0, 0);
+    for (const auto &f : regions) {
+        // The cost for pure inlined functions will be accounted in the
+        // consumer of the inlined function so they should be skipped.
+        if (inlines.find(f.first) != inlines.end()) {
+            internal_assert(get_element(env, f.first).is_pure());
+            continue;
+        }
+
+        Cost cost = region_cost(f.first, f.second, inlines);
+        if (cost.arith == unknown) {
+            return Cost(unknown, unknown);
+        } else {
+            total_cost.arith += cost.arith;
+            total_cost.memory += cost.memory;
+        }
+    }
+
+    internal_assert((total_cost.arith != unknown) && (total_cost.memory != unknown));
+    return total_cost;
+}
+
+namespace {
+
+/** Helper class that only accounts for the likely portion of the expression in
+ * the case of max, min, and select. This will help costing functions with
+ * boundary conditions better. The likely intrinsic triggers loop partitioning
+ * and on average (steady stage) the cost of the expression will be equivalent
+ * to the likely portion. */
+class LikelyExpression : public IRMutator {
+    using IRMutator::mutate;
+    using IRMutator::visit;
+
+    void visit(const Min *op) {
+        IRVisitor::visit(op);
+        bool likely_a = has_likely_tag(op->a);
+        bool likely_b = has_likely_tag(op->b);
+        if (likely_a && !likely_b) {
+            expr = op->a;
+        } else if (likely_b && !likely_a) {
+            expr = op->a;
+        }
+    }
+
+    void visit(const Max *op) {
+        IRVisitor::visit(op);
+        bool likely_a = has_likely_tag(op->a);
+        bool likely_b = has_likely_tag(op->b);
+        if (likely_a && !likely_b) {
+            expr = op->a;
+        } else if (likely_b && !likely_a) {
+            expr = op->b;
+        }
+    }
+
+    void visit(const Select *op) {
+        IRVisitor::visit(op);
+        bool likely_t = has_likely_tag(op->true_value);
+        bool likely_f = has_likely_tag(op->false_value);
+        if (likely_t && !likely_f) {
+            expr = op->true_value;
+        } else if (likely_f && !likely_t) {
+            expr = op->false_value;
+        }
+    }
+};
+
+} // anonymous namespace
+
+map<string, int64_t>
+RegionCosts::stage_detailed_load_costs(string func, int stage,
+                                       const set<string> &inlines) {
+    map<string, int64_t> load_costs;
+    Function curr_f = get_element(env, func);
+    Definition def = get_stage_definition(curr_f, stage);
+
+    for (const auto &e : def.values()) {
+        Expr inlined_expr = perform_inline(e, env, inlines);
+        // TODO: Handle likely.
+        //inlined_expr = LikelyExpression().mutate(inlined_expr);
```

Comment:
It's not dead code, but for now, we don't handle likely. I'll comment it out and add TODO.